src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx

```diff
@@ -392,12 +482,31 @@ export default class AppearanceUserSettingsTab extends React.Component<IProps, I
                     useCheckbox={true}
                     disabled={this.state.layout == Layout.IRC}
                 />
-                <StyledCheckbox
-                    checked={this.state.layout == Layout.IRC}
-                    onChange={(ev) => this.onIRCLayoutChange(ev.target.checked)}
-                >
-                    {_t("Enable experimental, compact IRC style layout")}
-                </StyledCheckbox>
+                { SettingsStore.getValue("feature_new_layout_switcher") ?
+                    <SettingsFlag
+                        name="singleSideBubbles"
+                        level={SettingLevel.DEVICE}
+                        useCheckbox={true}
+                        disabled={!(this.state.layout == Layout.Bubble) || this.state.adaptiveSideBubbles}
+                    /> : null
+                }
+                { SettingsStore.getValue("feature_new_layout_switcher") ?
+                    <SettingsFlag
+                        name="adaptiveSideBubbles"
+                        level={SettingLevel.DEVICE}
+                        useCheckbox={true}
+                        onChange={(checked) => this.setState({adaptiveSideBubbles: checked})}
+                        disabled={!(this.state.layout == Layout.Bubble)}
```

Comment:
```suggestion
                        disabled={this.state.layout !== Layout.Bubble}
```

Reply:
Also, does it make sense to also disable this when `singleSideBubbles` is enabled?

Reply:
No. If a default config enables both you might get stuck.

Reply:
`adaptiveSideBubbles` sort of overrides `singleSideBubbles`. That's why it's also fine semantically, I think.